tests/python/test_numpy.py

```diff
@@ -59,12 +59,16 @@ def test_advanced_decompression_checksum(self):
                 compress_param_num
             ),
         }
-        compressed_array = zfpy.compress_numpy(
+        compressed_array_t = zfpy.compress_numpy(
             random_array,
             write_header=False,
             **compression_kwargs
         )
-
+        if isinstance(compressed_array_t, np.ndarray):
```

Comment:
I'm not sure I understand the intent here.  `compressed_array_t` is by definition a `bytes` object, so this conditional should always be false.

Wouldn't the appropriate test be to ensure that we can pass both a `bytes` object and an `ndarray` to `zfpy.decompress_numpy`?  What other array-like objects are we trying to support here?

Minor nit: As a C/C++ programmer, I think `compressed_array_t` is a confusing name as the `_t` suffix makes it look like a type.